extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -271,33 +284,106 @@ public static boolean isYoutubeChannelMixId(@Nonnull final String playlistId) {
+     * Ids from a YouTube Genre Mix start with "RDGMEM"
      *
-     * @throws ParsingException If the playlistId is a Channel Mix or not a mix.
+     * @return Whether given id belongs to a YouTube Genre Mix
+     */
+    public static boolean isYoutubeGenreMixId(@Nonnull final String playlistId) {
+        return playlistId.startsWith("RDGMEM");
+    }
+
+    /**
+     * @param playlistId the playlist id to parse
+     * @return the {@link PlaylistInfo.PlaylistType} extracted from the playlistId (mix playlist
+     *         types included)
+     * @throws ParsingException if the playlistId is null or empty, if the playlistId is not a mix,
+     *                          if it is a mix but it's not based on a specific stream (this is the
+     *                          case for channel or genre mixes)
      */
     @Nonnull
-    public static String extractVideoIdFromMixId(@Nonnull final String playlistId)
+    public static String extractVideoIdFromMixId(final String playlistId)
             throws ParsingException {
-        if (playlistId.startsWith("RDMM")) { // My Mix
+        if (isNullOrEmpty(playlistId)) {
+            throw new ParsingException("Video id could not be determined from empty playlist id");
+
+        } else if (isYoutubeMyMixId(playlistId)) {
             return playlistId.substring(4);
 
-        } else if (isYoutubeMusicMixId(playlistId)) { // starts with "RDAMVM" or "RDCLAK"
+        } else if (isYoutubeMusicMixId(playlistId)) {
             return playlistId.substring(6);
 
-        } else if (isYoutubeChannelMixId(playlistId)) { // starts with "RMCM"
-            // Channel mix are build with RMCM{channelId}, so videoId can't be determined
-            throw new ParsingException("Video id could not be determined from mix id: "
+        } else if (isYoutubeChannelMixId(playlistId)) {
+            // Channel mixes are of the form RMCM{channelId}, so videoId can't be determined
+            throw new ParsingException("Video id could not be determined from channel mix id: "
+                    + playlistId);
+
+        } else if (isYoutubeGenreMixId(playlistId)) {
+            // Genre mixes are of the form RDGMEM{garbage}, so videoId can't be determined
+            throw new ParsingException("Video id could not be determined from genre mix id: "
                     + playlistId);
 
-        } else if (isYoutubeMixId(playlistId)) { // normal mix, starts with "RD"
+        } else if (isYoutubeMixId(playlistId)) { // normal mix
+            if (playlistId.length() != 13) {
+                // Stream YouTube mixes are of the form RD{videoId}, but if videoId is not exactly
+                // 11 characters then it can't be a video id, hence we are dealing with a different
+                // type of mix (e.g. genre mixes handled above, of the form RDGMEM{garbage})
+                throw new ParsingException("Video id could not be determined from mix id: "
```

Comment:
Do you have the 11 character number from that stackexchange post I shared? If yes one could link it here: https://webapps.stackexchange.com/questions/54443/format-for-id-of-youtube-video/101153#101153

Reply:
Even though I read your post, the fact that a youtube video id is 11 characters long was known to me. Anyway, I will add that.